core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java

```diff
@@ -1111,4 +1117,368 @@ public void testHashCodes() throws Exception {
     valueMetaString.type = ValueMetaInterface.TYPE_NONE;
     assertEquals( valueMetaString.hashCode( "any" ),  0 );
   }
+
+  @Test
+  public void testGetMetadataPreview() throws SQLException, KettleDatabaseException {
+    ValueMetaBase valueMetaBase = new ValueMetaBase( );
+
+    DatabaseMeta dbMeta = spy( new DatabaseMeta() );
+
+    // SQL CHAR is a Pentaho String
+    ResultSet resultSet = mock( ResultSet.class );
+    Mockito.when( resultSet.getInt( "DATA_TYPE" ) ).thenReturn( Types.CHAR );
+    ValueMetaInterface valueMeta = valueMetaBase.getMetadataPreview( dbMeta, resultSet );
+    Assert.assertTrue( valueMeta.isString() );
+
+    // SQL VARCHAR is a Pentaho String
```

Comment:
Please write a unit test for each test, something like:
```
@Test
public void sqlVarcharToPentahoString() {
  ...
}
```
If one of these tests fail you'll know exactly when is wrong, instead of having to debug a single test.
This way you can also avoid each comment before the test, the method name will be self explanatory.